pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -1216,6 +1226,50 @@ func (p *PravegaCluster) validateConfigMap() error {
 	return nil
 }
 
+// ValidateAuthenticationSettings checks for correct options passed to pravega
+// when authentication is enabled/disabled.
+func (p *PravegaCluster) ValidateAuthenticationSettings() error {
+	if p.Spec.Authentication.Enabled == true {
+		newkey, ok := p.Spec.Pravega.Options["autoScale.controller.connect.security.auth.enable"]
+		oldkey := ""
+		if !ok {
+			oldkey, ok = p.Spec.Pravega.Options["autoScale.authEnabled"]
+		}
+		if !ok {
+			return fmt.Errorf("autoScale.controller.connect.security.auth.enable field is not present")
+		} else if newkey == "false" || oldkey == "false" {
```

Comment:
Should we rather be checking instead that one (or both) of them is set to `"true"` and the other is unset (or both are set to `"true"`)?

Reply:
I am not sure if one of the keys is set to `true` and other is set to `false` which one will take precedence. I think second option will take precedence. But here .we are ensuring that both of the keys cannot be set to `false`.

Reply:
My concern was more about things like `"True"` or `"dummy"` (there is no validation)

Reply:
Made the changes to take care of that scenario and verified it. Please have a look